assets/global.js

```diff
@@ -443,6 +443,7 @@ class ModalDialog extends HTMLElement {
       'click',
       this.hide.bind(this, false)
     );
+    ['shopify:block:select', 'shopify:section:select'].forEach(event => document.addEventListener(event, () => this.hide()));
```

Comment:
But we need to access the `ModalDialog` and its `hide` method, and those are defined in here. How do I call that from `theme-editor.js` without importing it?

Reply:
Because we don't want to add lines of code in the final zip/code that the merchant's customers are viewing. As it's only applicable to the theme editor. So this specific behaviour where we're hiding the modals is editor specific for that custom class. 

But you can still reference functions declared for that custom element while adding the code in the `theme-editor.js` file. 
When you do a `querySelector` for that custom element, you can reference the functions that are tide to it 👌 Which is pretty awesome 🙂 
[Video example](https://screenshot.click/10-34-vczsv-vgyu1.mp4)

Reply:
Maybe it's something that can be brought up to the rest of the team. If we have an event existing when activating/deactivating the inspector, I don't see why we couldn't have one when the general settings or `App embeds` are clicked 🤔 
But I don't think it's something to worry about in this PR

Reply:
Oh, I'm sorry, I got Ludo a bit wrong. I thought he mentioned those events as already existing and we could easily use them but I didn't think they were the only ones we could use. I understand now.